### PR TITLE
Make Windows paths parsing more robust

### LINKED ISSUE #1885: Issue when reading csv file
Hello,

I have put some issues with dask/blaze for csv file reading (file can be read in pandas)...
Re-posting here:

When running this file here: 
https://goo.gl/xVOK4s

I got this message (even if tried the un-compressed csv file)
(I un gzip the file manually with winrar, it was ok.)

```
Traceback (most recent call last):
  File "D:\_devs\Python01\Anaconda27\lib\site-packages\IPython\core\formatters.py", line 222, in catch_format_error
    r = method(self, *args, **kwargs)
  File "D:\_devs\Python01\Anaconda27\lib\site-packages\IPython\core\formatters.py", line 699, in __call__
    printer.pretty(obj)
  File "D:\_devs\Python01\Anaconda27\lib\site-packages\IPython\lib\pretty.py", line 383, in pretty
    return _default_pprint(obj, self, cycle)
  File "D:\_devs\Python01\Anaconda27\lib\site-packages\IPython\lib\pretty.py", line 503, in _default_pprint
    _repr_pprint(obj, p, cycle)
  File "D:\_devs\Python01\Anaconda27\lib\site-packages\IPython\lib\pretty.py", line 694, in _repr_pprint
    output = repr(obj)
  File "D:\_devs\Python01\Anaconda27\lib\site-packages\blaze\interactive.py", line 444, in _choose_repr
    return expr_repr(self)
  File "D:\_devs\Python01\Anaconda27\lib\site-packages\blaze\interactive.py", line 354, in expr_repr
    return repr(coerce_scalar(compute(expr), str(expr.dshape)))
  File "D:\_devs\Python01\Anaconda27\lib\site-packages\multipledispatch\dispatcher.py", line 164, in __call__
    return func(*args, **kwargs)
  File "D:\_devs\Python01\Anaconda27\lib\site-packages\blaze\interactive.py", line 205, in compute
Out[7]:     return compute(expr, resources, **kwargs)
  File "D:\_devs\Python01\Anaconda27\lib\site-packages\multipledispatch\dispatcher.py", line 164, in __call__
    return func(*args, **kwargs)
  File "D:\_devs\Python01\Anaconda27\lib\site-packages\blaze\compute\core.py", line 394, in compute
    for e, dat in d2.items()
  File "D:\_devs\Python01\Anaconda27\lib\site-packages\blaze\compute\core.py", line 395, in <genexpr>
    if e in expr2
  File "D:\_devs\Python01\Anaconda27\lib\site-packages\multipledispatch\dispatcher.py", line 164, in __call__
    return func(*args, **kwargs)
  File "D:\_devs\Python01\Anaconda27\lib\site-packages\blaze\compute\csv.py", line 61, in pre_compute
    return dd.read_csv(data.path, **kwargs)
  File "D:\_devs\Python01\Anaconda27\lib\site-packages\dask\dataframe\csv.py", line 226, in read_csv
    **(storage_options or {}))
  File "D:\_devs\Python01\Anaconda27\lib\site-packages\dask\bytes\core.py", line 203, in read_bytes
    ensure_protocol(protocol)
  File "D:\_devs\Python01\Anaconda27\lib\site-packages\dask\bytes\core.py", line 387, in ensure_protocol
    raise ValueError("Unknown protocol %s" % protocol)
ValueError: Unknown protocol e
```

1) Do you have an idea how to read the file in Blaze ?

2) Also do you have regression tests on a seies of various real-life data file (even small size) to ensure that reading/writing various datatypes /files are fine (because I got another issue with HDF5 file writing before...,  it sounds in_out have some format reading/writing issues in Blaze...) ?

https://github.com/blaze/blaze/issues/1608

### PR DESCRIPTION
**Pull request by frol**
Fixes #1885.

/cc @mrocklin @arita37

### CODE REVIEW

File: dask/tests/test_utils.py [hunk 1]
```diff
@@ -178,9 +178,18 @@ def test_infer_storage_options():
...
+    (r'C:\\foo\bar', r'C:\\foo\bar'),
+    (r'c:/foo/bar', r'c:/foo/bar'),
+    (r'file:///c|\foo\bar', r'c:\foo\bar'),
+    (r'file:///C|/foo/bar', r'C:/foo/bar'),
+    (r'file:///C:/foo/bar', r'C:/foo/bar'),
+))
+def test_infer_storage_options_c(urlpath, expected_path):
+    so = infer_storage_options(urlpath)
     assert so['protocol'] == 'file'
+    assert so['path'] == expected_path
```
Comment (mrocklin) [inline dask/tests/test_utils.py]: Nice test!

**Comment by mrocklin** [comment]
Merging soon if no comments.

@frol thank you for always responding to these issues in a timely way.

**Comment by frol** [comment]
@mrocklin thank you for keeping such a beautiful project!

**Comment by frol** [comment]
Merge?

**Comment by arita37** [comment]
@frol   @mrocklin  :  Thx for the patch.

Just wondering (it might be stupid question):
     Suppose building /testing time gets longer/heavier (+ CPU process), just wondering 
    if there are tests on real input files (ie real life encoding issues on file) rather than code generated files ?

Another item/ idea (to enhance robustness through community testing): 

   Ability to upload zip containing : Script +  Data_file + explanation  into a folder/server...
   (either by python script or manually).
   Tests are run on the zip into this folder.   
   Logs file are generated for batch analysis.

Obvious points are : 
    Security of code executed.... (limit the import ??).
    Size of the zip file (data, limited).
    Template of script / output (platform, ...).

Main benefit is community can test for you in a controlled/template way, so less need to replicate the tests manually to identify the bug/issue.

By normalizing the process and on real data/life, bug discovery is also accelerated.
Tests on real life examples (although big files testing will be an issue of CPU/storage....).

Have a nice day.

**Comment by mrocklin** [comment]
Sounds like a fine idea to me @arita37 .  I would probably want files to be very small and would probably want scripts to be in the form of the pytest tests that we have today.  If you'd like to set up such a system and submit a PR that would be a good first step towards getting people's feedback on this.  I have no objections at the moment.

@frol  thanks for the fix.

**Pull request merged**